client/state/terms/actions.js

```diff
@@ -45,7 +45,6 @@ export function addTerm( siteId, taxonomy, term ) {
 					dispatch( receiveTerm( siteId, taxonomy, data ) );
 					return data;
 				},
-				() => Promise.resolve() // Silently ignore failure so we can proceed to remove temporary
```

Comment:
This will require us to update tests to expect an error to be thrown. 

Note that this might be tricky since it will be thrown on the underlying xhr object that the wpcom proxy request works with. 

I assume that that's why it was silenced like this. 

Perhaps in this case we can actually reject the promise and then handle it separately through a `.catch()`?